lib/state/simperium/middleware.ts

```diff
@@ -9,48 +11,88 @@ import * as T from '../../types';
 
 const debug = debugFactory('simperium-middleware');
 
-type Buckets = {
-  note: T.Bucket<T.Note>;
-};
-
-const buckets: Buckets = {} as Buckets;
+export const initSimperium = (
+  logout: () => any,
+  token: string,
+  username: string | null,
+  createWelcomeNote: boolean,
+  client: Client<'note' | 'preferences' | 'tag'>
+): S.Middleware => (store) => {
+  client.on('message', (message: string) => {
+    if (!message.startsWith('0:auth:')) {
+      return;
+    }
 
-export const storeBuckets = (newBuckets: Buckets) => {
-  buckets.note = newBuckets.note;
-};
+    const [prefix, authenticatedUsername] = message.split('0:auth:');
+    debug(`authenticated: ${authenticatedUsername}`);
 
-const fetchRevisions = (store: S.Store, state: S.State) => {
-  if (!state.ui.showRevisions || !state.ui.note) {
-    return;
+    if (username !== authenticatedUsername) {
+      debug(`was logged in as ${username} - logging out`);
+      return logout();
+    }
+  });
+
+  client.on('unauthorized', () => {
+    logout();
+  });
+
+  const noteBucket = client.bucket('note');
+
+  if (createWelcomeNote) {
+    import(
+      /* webpackChunkName: 'welcome-message' */ '../../welcome-message'
+    ).then(({ content }) => {
+      const now = Date.now() / 1000;
+      noteBucket.add({
+        content,
+        deleted: false,
+        systemTags: [],
+        creationDate: now,
+        modificationDate: now,
+        shareURL: '',
+        publishURL: '',
+        tags: [],
+      });
+    });
   }
 
-  const note = state.ui.note;
+  const fetchRevisions = (store: S.Store, state: S.State) => {
+    if (!state.ui.showRevisions || !state.ui.note) {
+      return;
+    }
 
-  buckets.note.getRevisions(
-    note.id,
-    (error: unknown, revisions: T.NoteEntity[]) => {
-      if (error) {
-        return debug(`Failed to load revisions for note ${note.id}: ${error}`);
-      }
+    const note = state.ui.note;
 
-      const thisState = store.getState();
-      if (!(thisState.ui.note && note.id === thisState.ui.note.id)) {
-        return;
-      }
+    noteBucket.getRevisions(
+      note.id,
+      (error: unknown, revisions: T.NoteEntity[]) => {
```

Comment:
is this truly unknown? You later use string interpolation. Can we assume it is not a string?

Reply:
it's unknown to me right now 😄 

the string interpolation relies on `toString()` which almost every type will have but that isn't itself meaningful enough for us do anything with. at some point I'd like an error audit, also a type definition for the Simperium library which would resolve this.

in the meantime `unknown` is serving like `any` but backwards - it prevents us from making assumptions about `error` in any way, which might keep us from raising type errors in runtime if we were to do something like `error.message` (which was causing type errors on sign-out and is something I fixed in this PR in the settings dialog).